monkey/tests/unit_tests/agent_plugins/exploiters/powershell/test_credentials_generator.py

```diff
@@ -0,0 +1,183 @@
+import sys
+from collections import namedtuple
+from copy import copy
+from typing import Optional, Sequence, Set, Type
+from unittest.mock import MagicMock
+
+import pytest
+from agent_plugins.exploiters.powershell.src.credentials_generator import (
+    generate_powershell_credentials,
+)
+from tests.data_for_tests.propagation_credentials import CREDENTIALS, IDENTITIES, SECRETS
+
+from common.credentials import Credentials, LMHash, NTHash, Password, Secret, Username
+
+TEST_DOMAIN = "TEST-DOMAIN"
+LOCAL_USER = "localuser"
```

Comment:
Should we have a test for if the local username is already in the propagation credentials? With/without the domain?